libautomata/src/main/java/com/mathewsachin/libautomata/ExitManager.kt

```diff
@@ -1,38 +1,41 @@
 package com.mathewsachin.libautomata
 
-import javax.inject.Inject
-import javax.inject.Singleton
+import kotlinx.coroutines.*
+import kotlin.time.Duration
 
 /**
  * Manages the termination of running scripts
  */
-@Singleton
-class ExitManager @Inject constructor() {
-    @Volatile
-    private var exitRequested = false
+class ExitManager {
+    private val scope = CoroutineScope(Dispatchers.IO)
+
+    fun wait(duration: Duration) {
+        checkExitRequested()
```

Comment:
Is checkExitRequested() now only done once before waiting?

Reply:
No, it's still being used before things like Region.exists, Region.findAll etc.

Reply:
I'm just not launching the coroutine if the scope is already cancelled.

Reply:
I meant that we don't manually check ourselves every second that someone stopped the script. But I just now read that scope.cancel() deals with terminating threads.

Reply:
yup, I misread your comment.
It also exits immediately unlike the once per second check.